### PR TITLE
get credentials from environment variables for clickhouse-client

### LINKED ISSUE #34538: Credentials from environment variables for `clickhouse-client`
The idea from @qoega 

**Use case**

Connect to your cloud ClickHouse instance easily.

**Describe the solution you'd like**

If user and/or password is not specified in command line or configuration file, get them from `CLICKHOUSE_USER` / `CLICKHOUSE_PASSWORD` environment variables.

If there is no such variable, it should work as before (with user `default` and empty password).

### PR DESCRIPTION
**Pull request by freedomDR**
Changelog category (leave one):
- New Feature

Changelog entry:
- When use clickhouse-client logining, If user and password is not specified in command line or configuration file, get them from `CLICKHOUSE_USER`  and `CLICKHOUSE_PASSWORD` environment variables. Close https://github.com/ClickHouse/ClickHouse/issues/34538

### CODE REVIEW

File: src/Client/ConnectionParameters.cpp [hunk 1]
```diff
@@ -57,6 +57,14 @@ ConnectionParameters::ConnectionParameters(const Poco::Util::AbstractConfigurati
             password = result;
     }
 
+    const char * env_user = getenv("CLICKHOUSE_USER");
+    const char * env_password = getenv("CLICKHOUSE_PASSWORD");
+    if (!config.has("user") && !config.has("password") && env_user && env_password)
```
Comment (vdimir) [inline src/Client/ConnectionParameters.cpp:62]: User may want to set just `CLICKHOUSE_USER` or `CLICKHOUSE_PASSWORD` but not both, so we can set it separately

**Review by vdimir — COMMENTED** [review]
It's good to add test, I suppose https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_user_zero_database_access/test_user_zero_database_access.py can be used.

File: src/Client/ConnectionParameters.cpp (hunk 1 above)
Comment (vdimir) [reply src/Client/ConnectionParameters.cpp:62]: Also values env should have higher priority than config, so set it even if config have some setup

Comment (freedomDR) [reply src/Client/ConnectionParameters.cpp:62]: So about priority, command user >  environment variables > config.xml config

Comment (vdimir) [reply src/Client/ConnectionParameters.cpp:62]: > So about priority, command user > environment variables > config.xml config

Yes

**Comment by freedomDR** [comment]
@vdimir  update.

**Comment by vdimir** [comment]
Let's make tests more specific to ensure that user is actually taken from env.
I've commited changes [`9e81bd3` (#34947)](https://github.com/ClickHouse/ClickHouse/pull/34947/commits/9e81bd395a566e7f4764d7efe7b47b93cffba605) but didn't checked it, let's wait CI

**Comment by freedomDR** [comment]
test_user_zero_database_access is passing. I checkout from clickhouse master, but I often see some failed test tasks.How to avoid some failed tests.

**Comment by vdimir** [comment]
> How to avoid some failed tests

Sometimes some checks broken on master, we'll fix it

**Review by vdimir — APPROVED** [review]

**Pull request merged**